design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,280 @@
+- [Issue description](#issue-description)
+- [Solution](#solution)
+    - [Azure Application Identity](#azure-application-identity)
+    - [Design](#design)
+        - [Authentication Flow](#authentication-flow)
+        - [Azure authenticator class](#azure-authenticator-class)
+    - [Backwards compatibility](#backwards-compatibility)
+    - [Performance](#performance)
+    - [Affected Components](#affected-components)
+- [Test plan](#test-plan)
+- [Logs](#logs)
+- [Docs](#docs)
+- [Version update](#version-update)
+- [DoD](#dod)
+- [Open Questions](#open-questions)
+
+## Glossary
+
+| Term                                   | Description                                                                                                                                                                                     |
+|----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Active Directory (AD)                  | an access management service that allows users to sign-in and access resources based on privileges granted to them                                                                              |
+| Active Directory (AD) tenant           | a dedicated instance of AD for the organization (AAM for example)                                                                                                                               |
+| Azure Resource Manager                 | the service that is used to provision resources. When creating a new VM, it receives a request upon instance (VM) creation and gives the instance a system-assigned managed identity.           |
+| Service principle                      | object representation of VM instance in AD. This object holds information defining its permissions on resources                                                                                 |
+| Client ID                              | unique identifier generated by Azure AD tied to application and service principal during initial provisioning. This client ID is used to identify the object that is requesting an access token |
+| System-assigned identity               | an identity for an instance in AD that is trusted. The identity is tied to the resource and abides to its lifecycle. In other words, if the resource is delete so to will the identity.         |
+| User-assigned identity                 | an identity for an instance in AD that is trusted. This identity can be assigned to 1+ Azure services instances and is independent of instance lifecycle                                        |
+| Azure Instance Metadata Service (IMDS) | a REST endpoint accessible to all VMs created via the Azure Resource Manager. Resources in Azure request access tokens from the Azure Instance Metadata service endpoint                        |
```

Comment:
Worth to mention that while, at first, we support VM, IMDS supports other services also, and we can leverage it too.